### PR TITLE
Camera software system

### PR DESCRIPTION
**Pull request by mrnye42**
Detailed Design Checkpoint 2. Good faith effort

### CODE REVIEW

File: Documentation/Signoffs/Camera_Software_System.md [hunk 1]
```diff
@@ -1,37 +1,87 @@
...
-
+## Functionality 
+The camera software system will be reponsible for controlling the camera hardware system, ensuring that the system is able to take an image of an unmanned aircraft system (UAS) in-flight or the UAS pilot and save it to the database for usage by campus police.
 ## Constraints
 | No.| Contraint | Origin |
 | -- | --------- |--------|
-|  1 |           |        |              
-|  2 |           |        |                          
-|  3 |           |        |              
+|  1 | The system shall not account for light levels and/or obstructions to the view of the UAS or control station when determining whether to take a picture | Stakeholder Constraint |
```
Comment (mrnye42) [inline Documentation/Signoffs/Camera_Software_System.md]: find the most relevant constraint then attach to the end. this is what will happen...but this wont happen.

Comment (mrnye42) [reply Documentation/Signoffs/Camera_Software_System.md]: light levels is a fuzzy term. may not be a software completely. the camera has its own light source so may need to say we wont do the flash in the hardware constraint

File: Documentation/Signoffs/Camera_Software_System.md [hunk 2]
```diff
@@ -1,37 +1,87 @@
...
+## Functionality 
+The camera software system will be reponsible for controlling the camera hardware system, ensuring that the system is able to take an image of an unmanned aircraft system (UAS) in-flight or the UAS pilot and save it to the database for usage by campus police.
 ## Constraints
 | No.| Contraint | Origin |
 | -- | --------- |--------|
-|  1 |           |        |              
-|  2 |           |        |                          
-|  3 |           |        |              
+|  1 | The system shall not account for light levels and/or obstructions to the view of the UAS or control station when determining whether to take a picture | Stakeholder Constraint | 
+|  2 | The system shall prioritize capturing a picture of the UAS user if the system obtains the location of the UAS user | Tech Police |
```
Comment (mrnye42) [inline Documentation/Signoffs/Camera_Software_System.md]: if not then will capture a picture of the the UAS

File: Documentation/Signoffs/Camera_Software_System.md [hunk 3]
```diff
@@ -1,37 +1,87 @@
 # Camera Software System
-## Functionality
-
+## Functionality 
+The camera software system will be reponsible for controlling the camera hardware system, ensuring that the system is able to take an image of an unmanned aircraft system (UAS) in-flight or the UAS pilot and save it to the database for usage by campus police.
 ## Constraints
 | No.| Contraint | Origin |
 | -- | --------- |--------|
-|  1 |           |        |
```
Comment (mrnye42) [inline Documentation/Signoffs/Camera_Software_System.md:7]: shall compute these pointing figures based on the data etc.

Comment (mrnye42) [inline Documentation/Signoffs/Camera_Software_System.md:7]: Another constraint: input shall be in the form of latitude, longitude, etc.

Comment (mrnye42) [reply Documentation/Signoffs/Camera_Software_System.md:7]: accuracy of the output of the computations constraint.

Comment (mrnye42) [reply Documentation/Signoffs/Camera_Software_System.md:7]: camera lense that is 30 deg can have more error

Comment (mrnye42) [reply Documentation/Signoffs/Camera_Software_System.md:7]: shall output picture in some format: binary probably. depends on hardware

Comment (mrnye42) [reply Documentation/Signoffs/Camera_Software_System.md:7]: raw file, or list a few standard formats

File: Documentation/Signoffs/Camera_Software_System.md (hunk 1 above)
Comment (mrnye42) [reply Documentation/Signoffs/Camera_Software_System.md]: contraint: will only succeed in predicting motion in these specific cases: straight line.  

accuracy constrain on output

opperating system constraint
how often do we have to output things to camera to keep things moving (analysis)

File: Documentation/Signoffs/Camera_Software_System.md [hunk 4]
```diff
@@ -1,37 +1,87 @@
...
+
+The code to determine the voltage that should be applied to the motor system so that the camera will face the UAS or control station will depend on the location of the UAS or control station, and the specs of the motor and camera as defined in the [Camera Hardware System](Camera_Hardware_System.md) detailed design markdown file.
+
+<img src= "/Documentation/Images/Camera_Software_System/Parsed_Flowchart_4.png" width="200" height="95"> 
+
+This second block with the instruction "determine whether a picture should be taken", will contain code that is only concerned with whether the UAS is still in view of the camera. This code will stop the system from taking a picture when the UAS or UAS user has moved out of scope of the camera in the time it takes the system to move the camera into the correct position. This block will utilize the information about the camera location and UAS or control station location as well as the specs of the motor and camera as defined in the [Camera Hardware System](Camera_Hardware_System.md) detailed design markdown file. This block will be followed by the instruction for the camera to take a picture.
+
+### Psuedo Code
+
```
Comment (mrnye42) [inline Documentation/Signoffs/Camera_Software_System.md:149]: math section: present equations to determine results for angles to point camera

Comment (mrnye42) [reply Documentation/Signoffs/Camera_Software_System.md:149]: analyze the processing power needed

Comment (mrnye42) [reply Documentation/Signoffs/Camera_Software_System.md:149]: CPU power

**Comment by ja3443** [comment]
Functionality
- need more detail as noted below
- what is meant by "controlling"? be specific about what is controlled (i.e. generate signals to control servos)
- if power or camera activation is controlled also mention that
- clarify that image is obtained from camera hardware, formatted, and sent to database

**Comment by ja3443** [comment]
Constraints
- 1 - change "shall not successfully" to "is not expected to" (the way it's written is that you are prohibited from doing that, even if you have x-ray vision)
- 3 - maybe change "but will not be performed" to "but is not required" to clarify that you don't have to do that but you can if you want to
- 3 and 4 - maybe these can be merged as they are addressing the same thing - need to define how straight is "straight" as mathematically they'll almost never line up due to measurement errors, etc.
- 6 - too vague - be specific about which standards and policies

**Comment by ja3443** [comment]
Flowchart and Analysis and BOM and References (brief review)
- the flowchart shows a polling system, not an event-based system - is this what is wanted? what happens if the system is waiting data from the database and it runs out of power? no alert will be issued
- move implementation details from analysis to flowchart / buildable schematic
- analysis section is where you show that what was described in the previous sections should work
- BOM - state "none" if there is nothing
- References - number 5 is a footnote, not a reference
- There is a public URL to access tech policy and that is a better choice to put in the references (will then need to select policy by number)
- math for pointing angles has errors

**Comment by ja3443** [comment]
Looks good. Merge approved. There is a "conflicting files" error that needs to be resolved before the merge can happen.

**Pull request merged**